software/control/widgets.py

```diff
@@ -7793,3 +7793,91 @@ def __init__(self, ObjectivesWidget, WellplateFormatWidget ,*args, **kwargs):
+        self.edit_text = QLineEdit(self)
+        self.edit_text.setMaxLength(1)  # Restrict to one character
+        self.edit_text.setText('0')
+        move_to_pos_label = QLabel("move to pos.", self)
+        self.move_spin_btn = QPushButton("Move To", self)
+
+        move_to_pos_layout = QHBoxLayout()
+        move_to_pos_layout.addWidget(move_to_pos_label)
+        move_to_pos_layout.addWidget(self.edit_text)
+        move_to_pos_layout.addWidget(self.move_spin_btn)
+
+        # Buttons for controlling the filter spin
+        self.previous_btn = QPushButton("Previous", self)
+        self.next_btn = QPushButton("Next", self)
+        self.home_btn = QPushButton("Homing", self)
+
+        buttons_layout = QHBoxLayout()
+        buttons_layout.addWidget(self.previous_btn)
+        buttons_layout.addWidget(self.next_btn)
+        buttons_layout.addWidget(self.home_btn)
+
+        # Main vertical layout
+        main_layout = QVBoxLayout()
+        main_layout.addLayout(position_layout)
+        main_layout.addLayout(status_layout)
+        main_layout.addLayout(move_to_pos_layout)  # Layout with editText, label, and button
+        main_layout.addLayout(buttons_layout)
+
+        self.setLayout(main_layout)
+
+        # Connect signals and slots
+        self.previous_btn.clicked.connect(self.decrement_position)
+        self.next_btn.clicked.connect(self.increment_position)
+        self.home_btn.clicked.connect(self.home_position)
+        self.move_spin_btn.clicked.connect(self.move_to_position)
+
+    def update_position(self, position):
+        self.position_label.setText(f"Position: {position}")
+
+    def decrement_position(self):
+        current_position = int(self.position_label.text().split(": ")[1])
+        new_position = max(0, current_position - 1)  # Ensure position doesn't go below 0
+        if current_position != new_position:
+            self.microscope.squid_filter_wheel.previous_position()
+            self.update_position(new_position)
+
+    def increment_position(self):
+        current_position = int(self.position_label.text().split(": ")[1])
+        new_position = min(7, current_position + 1)  # Ensure position doesn't go above 7
+        if current_position != new_position:
+            self.microscope.squid_filter_wheel.next_position()
+            self.update_position(new_position)
+
+    def home_position(self):
+        self.update_position(0)
+        self.status_label.setText("Status: Homed")
+        self.microscope.squid_filter_wheel.homing()
+
+    def move_to_position(self):
+        try:
+            position = int(self.edit_text.text())
+            if position != int(self.position_label.text().split(": ")[1]):
+                self.microscope.squid_filter_wheel.set_emission(position)
+            self.update_position(position)
+        except ValueError:
+            self.status_label.setText("Status: Invalid input")
+            self.position_label.setText("Position: Invalid")
```

Comment:
Is it right that the position ends up invalid in this case, or is it just unchanged from what it was before?

Reply:
The widget is for debugging squid filter code. So when user set the invalid value to the driver, we need only give the error message to user and keep the squid filter wheel at position it was before.